lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const logger = require('../../logger').child({ component: 'ElasticSearch' })
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
+  const actions = {
+    GET: 'get',
+    PUT: 'update',
```

Comment:
I know you said this was feature parity with Go agent but it reads weird especially on the segment of index create

Reply:
It does. I can revert, and we just note that it's going to diverge from them.